src/Microsoft.AspNet.Razor.Runtime/Runtime/TagHelpers/TagHelperExecutionContext.cs

```diff
@@ -224,18 +228,24 @@ public void AddTagHelperAttribute(string name, object value)
         }
 
         /// <summary>
-        /// Execute and retrieve the rendered child content asynchronously.
+        /// Execute children asynchronously with the given <paramref name="encoder"/> in scope and return their
+        /// rendered content.
         /// </summary>
+        /// <param name="useCachedResult">If <c>true</c> multiple calls to this method will not cause re-execution
+        /// of child content; cached content will be returned.</param>
+        /// <param name="encoder">
+        /// The <see cref="HtmlEncoder"/> to use. Does not override all HTML encodings which may occur.
+        /// </param>
         /// <returns>A <see cref="Task"/> that on completion returns the rendered child content.</returns>
         /// <remarks>
         /// Child content is only executed once. Successive calls to this method or successive executions of the
         /// returned <see cref="Task{TagHelperContent}"/> return a cached result.
         /// </remarks>
-        public async Task<TagHelperContent> GetChildContentAsync(bool useCachedResult)
+        public async Task<TagHelperContent> GetChildContentAsync(bool useCachedResult, HtmlEncoder encoder)
```

Comment:
> It'd be an inconsistency that we'd then change across the board to make passing null anywhere result in "do not encode".

Sorry I wasn't clear; was talking about the conversation up [here](https://github.com/aspnet/Razor/pull/653#discussion-diff-48931569).

In any case, there are two ways forward. @DamianEdwards and @NTaylorMullen, thoughts?

##### Use a `NullHtmlEncoder`
We or tag helper authors that are looking for no-op behaviour when calling `output.GetChildContentAsync` could create a relatively simple `NullHtmlEncoder`. We already have the [`HtmlTestEncoder`](https://github.com/aspnet/HtmlAbstractions/blob/dev/src/Microsoft.Extensions.WebEncoders/Testing/HtmlTestEncoder.cs) class in a production package (`Microsoft.Extensions.WebEncoders`) albeit in a special `...Testing` namespace. Implementation of `NullHtmlEncoder` would be similar.

This class (if we provide it) gives us an opportunity for further warnings about what will and won't be encoded, specifically around HTML-encoded values users type in directly. It also requires no changes to the approach shown in this PR and aspnet/MVC#3871 (except writing the class itself; again, if we provide it).

If we provide it, main questions on the class would be where to place it: Which namespace and which package? I'd suggest `System.Text.Encodings.Web` and the `Microsoft.AspNet.Razor.Runtime` package (i.e. same package as `TagHelperOutput`).

##### Special-case a `null` `HtmlEncoder` value
Requires rework of this PR to make an accessor for the default `HtmlEncoder` available in `TagHelperOutput`. Requirement would bubble up through `TagHelperRunner` and `TagHelperExecutionContext` into `TagHelperScopeManager`. From there would require generation changes for either the `TagHelperScopeManager` instantiation or all `TagHelperScopeManager.Begin()` calls. (First is a little more straightforward since the accessor never needs to change. But should probably be consistent with handling of `RazorPage.StartTagHelperWritingScope()` and `RazorPage.EndTagHelperWritingScope()`.)

Also requires changes to _every_ `IHtmlContent.WriteTo()` implementation across the HtmlAbstractions, Razor and MVC repos. Need to both remove `ArgumentNullException` cases and do something different when the passed encoder is `null`. This is a significant contract change that can't be enforced in 3rd party `IHtmlContent` implementations i.e. they'll break long after, when first used with a tag helper that calls `output.GetChildContentAsync(encoder: null)`.

Note I'm not whining this approach is lots of work (ok, not only whining). It's a lot for a small scenario but the contract changes are a larger concern. In addition `null` normally means "default" in our APIs, not "do this special other thing".